src/pkcs11/pkcs11-spy.c

```diff
@@ -400,6 +400,114 @@ spy_attribute_list_out(const char *name, CK_ATTRIBUTE_PTR pTemplate,
 	print_attribute_list(spy_output, pTemplate, ulCount);
 }
 
+static void
+spy_dump_mechanism(const char *name, CK_MECHANISM_PTR pMechanism)
+{
+	if (!pMechanism) {
+		fprintf(spy_output, "%s=(null)\n", name);
```

Comment:
Here, we should print the `[in] ` prefix to keep it in sync with other parameters dumps. The same for the following prints below, where the prefix is missing.

Reply:
OK